syft/pkg/alpm_metadata.go

```diff
@@ -0,0 +1,83 @@
+package pkg
+
+import (
+	"sort"
+	"time"
+
+	"github.com/anchore/packageurl-go"
+	"github.com/anchore/syft/syft/file"
+	"github.com/anchore/syft/syft/linux"
+	"github.com/scylladb/go-set/strset"
+)
+
+const AlpmDBGlob = "**/var/lib/pacman/local/**/desc"
+
+type AlpmMetadata struct {
+	BasePackage  string           `mapstructure:"base" json:"basepackage"`
+	Package      string           `mapstructure:"name" json:"package"`
+	Version      string           `mapstructure:"version" json:"version"`
+	Description  string           `mapstructure:"desc" json:"description"`
+	Architecture string           `mapstructure:"arch" json:"architecture"`
+	Size         int              `mapstructure:"size" json:"size" cyclonedx:"size"`
+	Packager     string           `mapstructure:"packager" json:"packager"`
+	License      string           `mapstructure:"license" json:"license"`
+	URL          string           `mapstructure:"url" json:"url"`
+	Validation   string           `mapstructure:"validation" json:"validation"`
+	Reason       int              `mapstructure:"reason" json:"reason"`
+	Files        []AlpmFileRecord `mapstructure:"files" json:"files"`
+	Backup       []AlpmFileRecord `mapstructure:"backup" json:"backup"`
+}
+
+// TODO: Implement mtree support
+type AlpmFileRecord struct {
+	Path         string       `mapstruture:"path" json:"path,omitempty"`
+	Type         string       `mapstructure:"type" json:"type,omitempty"`
+	UID          string       `mapstructure:"uid" json:"uid,omitempty"`
+	GID          string       `mapstructure:"gid" json:"gid,omitempty"`
+	Time         time.Time    `mapstructure:"time" json:"time,omitempty"`
+	Size         string       `mapstructure:"size" json:"size,omitempty"`
+	Md5digest    string       `mapstructure:"md5digest" json:"md5digest,omitempty"`
+	Sha256digest string       `mapstructure:"sha256digest" json:"sh256digest,omitempty"`
+	Link         string       `mapstructure:"link" json:"link,omitempty"`
+	Digest       *file.Digest `json:"digest,omitempty"`
```

Comment:
providing `Md5digest` + `Sha256digest` as well as `Digest` seems a little odd. Can we normalize this by removing the `Md5digest`, `Sha256digest`, and `Digest` fields and replacing them with a single `Digests []file.Digest` field?

Reply:
I'd remove `Md5digest` and `Sha256digest` and just stick with the singular `Digest` object. One file record should only have one digest. The complexity comes from `mtree` using `sha256` and the backup files in the files database from `alpm` using md5.

You would just need to remove the line here I believe?

https://github.com/anchore/syft/pull/943/files/dcb4c58d35e60cfc7333333023197579d6f47fcc#diff-f43151a6c3a5eef143a2ea5ceea35edc464d00e243d34365bdda7f6a4bb0b209R113

Reply:
that makes sense, thanks 👍